### PR TITLE
feat(crafters): add AlbumentationsCrafter

### LINKED ISSUE #1907: Add AlbumentationsCrafter
I propose that we add an [Albumentations](https://github.com/albumentations-team/albumentations) crafter, which could supplant/replace (with proper depreciation strategy) a lot of current image crafters, namely `CenterImageCropper`, `ImageCropper`, `ImageFlipper`, `ImageNormalizer`, `ImageResizer` and `RandomImageCropper`.

Here's are the reasons I believe this would be a good choice:
1. We replace these Jina "bespoke" crafters, with something that is [state-of-the-art in performance](https://github.com/albumentations-team/albumentations#benchmarking-results), [widely used](https://github.com/albumentations-team/albumentations#who-is-using-albumentations) and [extremely well documented](https://albumentations.ai/docs/).
2. Maintnance costs (in terms of labor) go down dramatically for these crafters :)
3. This not only replaces the current crafters, but adds 10s of new transform options, right out of the box.
4. We can implement chaining of transforms within the same crafter. Right now, to apply 2 transforms to an image, 2 crafters need to be added to the flow, and the images are passed between them. This adds a lot of unnecessary overhead. Albumentations supports effortless chaining of transforms, like this:
  ``` python
  transform = A.Compose([
    A.RandomCrop(width=256, height=256),
    A.HorizontalFlip(p=0.5),
    A.RandomBrightnessContrast(p=0.2),
])
  ```

In AlbumentationsCrafter, this could be done like this:
``` python
AlbumentationsCrafter(transforms = [
    {"RandomCrop": {"width": 256, "height":256}},
    {"HorizontalFlip": {"p": 0.5}},
    {"RandomBrightnessContrast": {"p": 0.2}}
])
```
I think this could also be used in `yaml` using the `with` argument? It's just a list/dict, nothing special here.
The object instatiation based on class name is also easy, [Hydra does it](https://hydra.cc/docs/next/patterns/instantiate_objects/overview/), for example.

If people think this is a good idea I could contribute a PR, and work on identifying examples where this new crafter could replace the existing ones :)

### PR DESCRIPTION
**Pull request by tadejsv**
This adds AlbumentationsCrafter. 
Closes #1907

### CODE REVIEW

**Comment by tadejsv** [comment]
@JoanFM this is ready for review, although I will improve the readme tomorrow. Also, build fails due to some Jina problem, as all tests pass.

**Review by JoanFM — COMMENTED** [review]
The problem is that we need some default value for the input arguments, because we do some testing to see if the image can be loaded in a Flow and in this case it will raise an Exception.

File: crafters/image/AlbumentationsCrafter/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+jina
```
Comment (JoanFM) [inline crafters/image/AlbumentationsCrafter/requirements.txt:1]: please remove jina as the image is based om jina

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by tadejsv** [comment]
@JoanFM any idea why the build is failing now? I've addressed the previous issues. Also, this PR is now complete

**Review by JoanFM — DISMISSED** [review]

File: crafters/image/AlbumentationsCrafter/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+manifest_version: 1
+name: AlbumentationsCrafter
+kind: crafter
+description: Apply transformations from the Albumentations package to the image.
+author: Jina AI Dev-Team (dev-team@jina.ai)
```
Comment (JoanFM) [inline crafters/image/AlbumentationsCrafter/manifest.yml:5]: Hey @tadejsv , is not the root of the problem, so no big deal, but you are welcome to attribute yourself as the author of this Hub Executor

**Comment by JoanFM** [comment]
Hey @tadejsv , I have tried locally, and it seems to be your `Dockerfile` that is not being properly built from `jina-hub`. I would suggest to move it to the old way, and open an issue in `jina-core` to have it solved

**Comment by tadejsv** [comment]
@JoanFM also (probably) not the root of the problem, but when exploring the URL of the docker request, it seems that jina inserts labels after every `FROM` line in the dockerfile. The correct way would be to do it only after the last `FROM`

Here's the decoded URL:
```
docker://localhost/v1.40/build?t=jinahub/pod.crafter.albumentationscrafter:0.0.1-0.9.18&q=False&nocache=False&rm=True&forcerm=False&pull=True&dockerfile=.dockerfile.85ba69d05f9914e2aef3438da3fd8b99512ce32e&dockerfile=FROM jinaai/jina AS base
LABEL ai.jina.hub.name="AlbumentationsCrafter" \      
ai.jina.hub.description="Apply transformations from the Albumentations package to the image." \      
ai.jina.hub.type="pod" \      
ai.jina.hub.author="Tadej Svetina (@tadejsv)" \      
ai.jina.hub.url="https://jina.ai" \      
ai.jina.hub.documentation="https://github.com/jina-ai/jina-hub" \      
ai.jina.hub.version="0.0.1" \      
ai.jina.hub.vendor="Jina AI Limited" \      
ai.jina.hub.license="apache-2.0" \      
ai.jina.hub.avatar="None" \      
ai.jina.hub.platform="['linux/amd64']" \      
ai.jina.hub.keywords="['image', 'transform', 'augmentation', 'albumentations']" \      
ai.jina.hub.manifest_version="1" \      
ai.jina.hub.kind="crafter" \      
ai.jina.hub.jina_version="0.9.18"
# setup the workspace
COPY . /workspace
WORKDIR /workspace

# install the third-party requirements
RUN pip install -r requirements.txt --no-cache-dir

# for testing the image
FROM base
LABEL ai.jina.hub.name="AlbumentationsCrafter" \      
ai.jina.hub.description="Apply transformations from the Albumentations package to the image." \      
ai.jina.hub.type="pod" \      
ai.jina.hub.author="Tadej Svetina (@tadejsv)" \      
ai.jina.hub.url="https://jina.ai" \      
ai.jina.hub.documentation="https://github.com/jina-ai/jina-hub" \      
ai.jina.hub.version="0.0.1" \      
ai.jina.hub.vendor="Jina AI Limited" \      
ai.jina.hub.license="apache-2.0" \      
ai.jina.hub.avatar="None" \      
ai.jina.hub.platform="['linux/amd64']" \      
ai.jina.hub.keywords="['image', 'transform', 'augmentation', 'albumentations']" \      
ai.jina.hub.manifest_version="1" \      
ai.jina.hub.kind="crafter" \      
ai.jina.hub.jina_version="0.9.18"RUN pip install pytest && pytest

FROM base
LABEL ai.jina.hub.name="AlbumentationsCrafter" \      
ai.jina.hub.description="Apply transformations from the Albumentations package to the image." \      
ai.jina.hub.type="pod" \      
ai.jina.hub.author="Tadej Svetina (@tadejsv)" \      
ai.jina.hub.url="https://jina.ai" \      
ai.jina.hub.documentation="https://github.com/jina-ai/jina-hub" \      
ai.jina.hub.version="0.0.1" \      
ai.jina.hub.vendor="Jina AI Limited" \      
ai.jina.hub.license="apache-2.0" \      
ai.jina.hub.avatar="None" \      
ai.jina.hub.platform="['linux/amd64']" \      
ai.jina.hub.keywords="['image', 'transform', 'augmentation', 'albumentations']" \      
ai.jina.hub.manifest_version="1" \      
ai.jina.hub.kind="crafter" \      
ai.jina.hub.jina_version="0.9.18"ENTRYPOINT ["jina", "pod", "--uses", "config.yml"]
```

**Comment by JoanFM** [comment]
> @JoanFM also (probably) not the root of the problem, but when exploring the URL of the docker request, it seems that jina inserts labels after every `FROM` line in the dockerfile. The correct way would be to do it only after the last `FROM`
> 
> Here's the decoded URL:
> 
> ```
> docker://localhost/v1.40/build?t=jinahub/pod.crafter.albumentationscrafter:0.0.1-0.9.18&q=False&nocache=False&rm=True&forcerm=False&pull=True&dockerfile=.dockerfile.85ba69d05f9914e2aef3438da3fd8b99512ce32e&dockerfile=FROM jinaai/jina AS base
> LABEL ai.jina.hub.name="AlbumentationsCrafter" \      
> ai.jina.hub.description="Apply transformations from the Albumentations package to the image." \      
> ai.jina.hub.type="pod" \      
> ai.jina.hub.author="Tadej Svetina (@tadejsv)" \      
> ai.jina.hub.url="https://jina.ai" \      
> ai.jina.hub.documentation="https://github.com/jina-ai/jina-hub" \      
> ai.jina.hub.version="0.0.1" \      
> ai.jina.hub.vendor="Jina AI Limited" \      
> ai.jina.hub.license="apache-2.0" \      
> ai.jina.hub.avatar="None" \      
> ai.jina.hub.platform="['linux/amd64']" \      
> ai.jina.hub.keywords="['image', 'transform', 'augmentation', 'albumentations']" \      
> ai.jina.hub.manifest_version="1" \      
> ai.jina.hub.kind="crafter" \      
> ai.jina.hub.jina_version="0.9.18"
> # setup the workspace
> COPY . /workspace
> WORKDIR /workspace
> 
> # install the third-party requirements
> RUN pip install -r requirements.txt --no-cache-dir
> 
> # for testing the image
> FROM base
> LABEL ai.jina.hub.name="AlbumentationsCrafter" \      
> ai.jina.hub.description="Apply transformations from the Albumentations package to the image." \      
> ai.jina.hub.type="pod" \      
> ai.jina.hub.author="Tadej Svetina (@tadejsv)" \      
> ai.jina.hub.url="https://jina.ai" \      
> ai.jina.hub.documentation="https://github.com/jina-ai/jina-hub" \      
> ai.jina.hub.version="0.0.1" \      
> ai.jina.hub.vendor="Jina AI Limited" \      
> ai.jina.hub.license="apache-2.0" \      
> ai.jina.hub.avatar="None" \      
> ai.jina.hub.platform="['linux/amd64']" \      
> ai.jina.hub.keywords="['image', 'transform', 'augmentation', 'albumentations']" \      
> ai.jina.hub.manifest_version="1" \      
> ai.jina.hub.kind="crafter" \      
> ai.jina.hub.jina_version="0.9.18"RUN pip install pytest && pytest
> 
> FROM base
> LABEL ai.jina.hub.name="AlbumentationsCrafter" \      
> ai.jina.hub.description="Apply transformations from the Albumentations package to the image." \      
> ai.jina.hub.type="pod" \      
> ai.jina.hub.author="Tadej Svetina (@tadejsv)" \      
> ai.jina.hub.url="https://jina.ai" \      
> ai.jina.hub.documentation="https://github.com/jina-ai/jina-hub" \      
> ai.jina.hub.version="0.0.1" \      
> ai.jina.hub.vendor="Jina AI Limited" \      
> ai.jina.hub.license="apache-2.0" \      
> ai.jina.hub.avatar="None" \      
> ai.jina.hub.platform="['linux/amd64']" \      
> ai.jina.hub.keywords="['image', 'transform', 'augmentation', 'albumentations']" \      
> ai.jina.hub.manifest_version="1" \      
> ai.jina.hub.kind="crafter" \      
> ai.jina.hub.jina_version="0.9.18"ENTRYPOINT ["jina", "pod", "--uses", "config.yml"]
> ```

it could very much be the root of the problem, you can please add it as a hint when opening the issue

**Comment by tadejsv** [comment]
Ok, found the actual problem: after labels jina does not insert (perserve) the linebreak. Will raise it as an issue

**Comment by tadejsv** [comment]
The CI is now passing as I added line breaks after the `FROM` statement, which circumvents the issue.

**Review by JoanFM — APPROVED** [review]

**Comment by JoanFM** [comment]
Hey @tadejsv ,

Thank you very much for yet another great contribution!

**Pull request merged**